plugin/pkg/auth/authorizer/rbac/bootstrappolicy/controller_policy.go

```diff
@@ -188,7 +188,8 @@ func init() {
 			// glusterfs
 			rbac.NewRule("get", "list", "watch").Groups(storageGroup).Resources("storageclasses").RuleOrDie(),
 			rbac.NewRule("get", "create", "delete").Groups(legacyGroup).Resources("services", "endpoints").RuleOrDie(),
-			rbac.NewRule("get").Groups(legacyGroup).Resources("secrets").RuleOrDie(),
+			// glusterfs and Azure File
+			rbac.NewRule("create", "get").Groups(legacyGroup).Resources("secrets").RuleOrDie(),
```

Comment:
> @smarterclayton 
>  Or are you saying the only way Azure file provisioning works is for the node to have highly privileged credentials (mount any file) and to do the operation from the node?

> @rootfs 
> azure file is a shared filesystem (cifs), it doesn't implement attach/detach.
> The secret is needed to mount the cifs share

doesn't the node have the azure cloud provider as well? why can it not get the storage key directly from the azure storage API?